src/server/Mine.cpp

```diff
@@ -0,0 +1,29 @@
+#include <stdio.h>
+#include <vector>
+#include <memory>
+#include <iostream>
+#include <numeric>
+#include "Mine.hpp"
+#include "../common/collision.hpp"
+#include "../common/game_settings.hpp"
+
+Mine::Mine(unsigned int id, sf::Vector2f position, float size)
+    :CircleGameObject(id, position, size) {}
+
+Mine::~Mine() {}
+
+std::vector<std::shared_ptr<Mine>> Mine::getActiveMines(std::vector<std::shared_ptr<Mine>>& mines) {
+    std::vector<std::shared_ptr<Mine>> active_mines;
+    std::copy_if(
+        mines.begin(), mines.end(), std::back_inserter(active_mines),
+        [](std::shared_ptr<Mine> mine){return mine->isActive();});
+    return active_mines;
+}
+
+std::vector<std::shared_ptr<Circle>> Mine::getCircles(std::vector<std::shared_ptr<Mine>>& mines) {
```

Comment:
`Mine::getCircles` and `Group::getCircles` can probably be moved to `CircleGameObject`

Reply:
Not going to do this because once we create a Physics World this will be irrelevant - we won't need to get the `Circle`s from the `GameObject`s, instead the Physics World will own the rigid bodies and iterate over them.